includes/Modules/Analytics_4/Web_Tag.php

```diff
@@ -43,12 +62,39 @@ public function register() {
 	protected function enqueue_gtag_script() {
 		if ( did_action( 'googlesitekit_analytics_init_tag' ) ) {
 			// If the gtag script is already registered in the Analytics module, then we need to add <MEASUREMENT_ID> configuration only.
-			$config = sprintf( 'gtag("config", "%s");', esc_js( $this->tag_id ) );
-			wp_add_inline_script( 'google_gtagjs', $config );
+			$this->add_inline_config(
+				$this->tag_id,
+				! empty( $this->custom_dimensions )
+					? $this->custom_dimensions
+					: array()
+			);
 		} else {
 			// Otherwise register gtag as in the Analytics module knowing that we used Measurement ID from GA4 instead of Property ID.
 			parent::enqueue_gtag_script();
 		}
 	}
 
+	/**
+	 * Gets the tag config as used in the gtag data vars.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Tag configuration.
+	 */
+	protected function get_tag_config() {
+		$config = parent::get_tag_config();
+
+		// Do not add custom dimensions if UA is enabled because they will be
+		// added to the UA property instead of to the GA4 measurement ID.
+		if ( did_action( 'googlesitekit_analytics_init_tag' ) ) {
```

Comment:
Hey @eugene-manuilov, thanks for making the change - however, I'm not sure if I am missing something, but I don't see that we're actually invoking this `googlesitekit_analytics_init_tag` action. Maybe it had a `do_action()` that's been removed and the hook is no longer in play?

Reply:
@techanvil we call it here: https://github.com/google/site-kit-wp/blob/019635e605e68078aff6fc6f7c1d224a75e1349a/includes/Core/Modules/Tags/Module_Web_Tag.php#L102

Reply:
Thanks for clarifying, @eugene-manuilov! I should have looked harder :facepalm: